From 788aff7e7416391a0de225b52a19b352a7c2bc5e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 26 May 2025 14:00:23 +0300
Subject: [PATCH] [SW-0] fs: boot_desc: cosmetic changes

Remove the useless internal nameless struct notion.
Update the ascii art structures with correct data.
No functional changes.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
Change-Id: I3430a32216a999fb83546f5584cc2b8ad8925a8c
---

diff --git a/fs/common/hl_boot_desc.h b/fs/common/hl_boot_desc.h
index 9e4a7fa..0ecd68b 100644
--- a/fs/common/hl_boot_desc.h
+++ b/fs/common/hl_boot_desc.h
@@ -97,35 +97,36 @@
 
 /*
  *           struct hl_fw_boot_ptrs                     struct hl_fw_img_desc
- *           =======================                       =================
- *          || Boot_ptrs_img_sz     ||              +---->|| Magic         ||
- *          | ---------------------- |              |     | --------------- |
- *          || Img_desc_sz          ||              |     || Version       ||
- *          | ---------------------- |              |     | --------------- |
- *          || Magic                ||              |     || Index         ||
- *          | ---------------------- |              |     | --------------- |
- *          || Count                ||              |     || Offset    [2] ||
- *          | ---------------------- |              |     | --------------- |
- *          || Pub key sz           ||              |     || Size      [2] ||
- *          | ---------------------- |              |     | --------------- |
- *          || Sig sz               ||              |     || Load High [2] ||
- *          | ---------------------- |              |     | --------------- |
- *          || Sec ver              ||              |     || Load Low  [2] ||
- *          | ---------------------- |              |     | --------------- |
- *          || Key idx              ||              |     || CRC32     [2] ||
- *          | ---------------------- |              |     | --------------- |
- *          || version              ||              |     || SHA384    [2] ||
- *          | ---------------------- |              |     | --------------- |
- *          || boot_ptrs_sz         ||              |     || Start High [2 ||
- *          | ---------------------- |              |     | --------------- |
- *     +--<<|| Offset2array         ||              |     || Start Low [2] ||
- *     |      ======================                |     |=================|
- *     |                                            |
- *     |                                            |
- *     |                                            |
- *     |                                            |
- *     |    Following Array                         |
- *     |     ======================                 |
+ *            ======================                        =================
+ *          || Boot_ptrs_img_sz     ||              +---->|| Magic           ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || Img_desc_sz          ||              |     || Version         ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || Magic                ||              |     || Offset      [2] ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || Count                ||              |     || Size            ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || Pub key sz           ||              |     || Load H/L        ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || Sig sz               ||              |     || CRC32           ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || Sec ver              ||              |     || SHA384          ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || Key idx              ||              |     || Start H/L       ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || version              ||              |     || DMA DST H/L [2] ||
+ *          | ---------------------- |              |     | ----------------- |
+ *          || boot_ptrs_sz         ||              |     || FW COMP VER []  ||
+ *          | ---------------------- |              |     | ----------------- |
+ *     +--<<|| Offset2array         ||              |     || CPU CFG BASE    ||
+ *     |      ======================                |     | ----------------- |
+ *     |                                            |     || MEM BASE H/L    ||
+ *     |                                            |     | ----------------- |
+ *     |                                            |     || MEM SIZE        ||
+ *     |                                            |     | ----------------- |
+ *     |                                            |     || FLAGS           ||
+ *     |    Following Array                         |       =================
+ *     |      ======================                |
  *     +--->|| Offset               ||>>------------+
  *     |    | ---------------------- |                     =================
  *     +--->|| Offset               ||>>----------------->|  ...            |
@@ -133,29 +134,27 @@
  *     :       :                         :                    :
  *     |    | ---------------------- |                     =================
  *     +--->|| Offset               ||>>----------------->|  ...            |
- *           ========================                      =================
+ *            ======================                       =================
  *
  */
 
 struct hl_fw_boot_ptrs {
-	struct {
-		/*
-		 * boot pointers image size
-		 * sizeof(struct hl_fw_boot_ptrs) +
-		 * sizeof(offset to descriptor)*count +
-		 * sizeof(hl_fw_img_desc)*count +
-		 * key +
-		 * signature size +
-		 * alignment (0-15 bytes) +
-		 * CRC (4 Bytes)
-		 */
-		uint32_t boot_ptrs_img_sz;
-		/*
-		 * single image descriptor size
-		 * sizeof(struct hl_fw_img_desc)
-		 */
-		uint32_t img_desc_sz;
-	};
+	/*
+	 * boot pointers image size
+	 * sizeof(struct hl_fw_boot_ptrs) +
+	 * sizeof(offset to descriptor)*count +
+	 * sizeof(hl_fw_img_desc)*count +
+	 * key +
+	 * signature size +
+	 * alignment (0-15 bytes) +
+	 * CRC (4 Bytes)
+	 */
+	uint32_t boot_ptrs_img_sz;
+	/*
+	 * single image descriptor size
+	 * sizeof(struct hl_fw_img_desc)
+	 */
+	uint32_t img_desc_sz;
 	struct hl_fw_boot_magic magic;
 	uint32_t count;
 	uint32_t pub_key_sz;	/* Public key size */
